libs/sfm/bundler_common.cc

```diff
@@ -266,5 +266,56 @@ load_survey_from_file (std::string const& filename,
     in.close();
 }
 
+/* ---------------------- Feature undistortion -------------------- */
+
+namespace {
+    double distort_squared_radius (double const r2, double const k1,
+        double const k2)
+    {
+        // Compute the distorted squared radius:
+        // (radius * distortion_coeff)^2 = r^2 * coeff^2
+        double coeff = 1.0 + r2 * k1 + r2 * r2 * k2;
+        return r2 * coeff * coeff;
+    }
+
+    double solve_undistorted_squared_radius (double const r2,
+        double const k1, double const k2)
+    {
+        // We use bisection as we can easily find a pretty good initial guess,
+        // and we don't want to compute all roots for a 5th degree polynomial.
+
+        // Guess initial interval upper and lower bound
+        double lbound = r2, ubound = r2;
+        while (distort_squared_radius(lbound, k1, k2) > r2)
+            lbound /= 1.05;
+        while (distort_squared_radius(ubound, k1, k2) < r2)
+            ubound *= 1.05;
+
+        // Perform a bisection until epsilon accuracy is reached
+        while (std::numeric_limits<double>::epsilon() < ubound - lbound)
```

Comment:
you need std::abs()?

Reply:
The upper bound is always bigger than the lower bound.